src/utils/helpers/number.test.ts

```diff
@@ -32,195 +32,220 @@ describe('Test helpers', () => {
+        expect(
+          prettifyNumber(1234567, { abbreviate: true, currentCurrency: 'USD' })
+        ).toEqual('$1.2M');
+      });
 
-    test('should return "$1.2M" for input 1234567, with abbreviate true, and currentCurrency is USD', () => {
-      expect(
-        prettifyNumber(1234567, { abbreviate: true, currentCurrency: 'USD' })
-      ).toEqual('$1.2M');
-    });
+      test('should return "1.2M ETH" for input 1234567, with abbreviate true, and currentCurrency is ETH', () => {
+        expect(
+          prettifyNumber(1234567, { abbreviate: true, currentCurrency: 'ETH' })
+        ).toEqual('1.2M ETH');
+      });
 
-    test('should return "1.2M ETH" for input 1234567, with abbreviate true, and currentCurrency is ETH', () => {
-      expect(
-        prettifyNumber(1234567, { abbreviate: true, currentCurrency: 'ETH' })
-      ).toEqual('1.2M ETH');
+      test('should return "¥1.2M" for input 1234567, with abbreviate true, and currentCurrency is JPY', () => {
+        expect(
+          prettifyNumber(1234567, { abbreviate: true, currentCurrency: 'JPY' })
+        ).toEqual('¥1.2M');
+      });
     });
 
-    test('should return "¥1.2M" for input 1234567, with abbreviate true, and currentCurrency is JPY', () => {
-      expect(
-        prettifyNumber(1234567, { abbreviate: true, currentCurrency: 'JPY' })
-      ).toEqual('¥1.2M');
-    });
+    describe('Numbers', () => {
+      test('should return "0" for input 0 and no currentCurrency selected', () => {
+        expect(prettifyNumber(0)).toEqual('0');
+      });
 
-    test('should return "1.2M" for input 1234567 and no currentCurrency selected', () => {
-      expect(prettifyNumber(1234567, { abbreviate: true })).toEqual('1.2 m');
-    });
+      test('should return "1.2M" for input 1234567 and no currentCurrency selected', () => {
+        expect(prettifyNumber(1234567, { abbreviate: true })).toEqual('1.2 m');
+      });
 
-    test('should return "1.234567" for input 1.23456789 and no currentCurrency selected', () => {
-      expect(prettifyNumber(1.23456789, { highPrecision: true })).toEqual(
-        '1.234567'
-      );
-    });
+      test('should return "1.234567" for input 1.23456789 and no currentCurrency selected', () => {
+        expect(prettifyNumber(1.23456789, { highPrecision: true })).toEqual(
+          '1.234567'
+        );
+      });
 
-    test('should return "< 0.000001" for input 0.0000001 and no currentCurrency selected', () => {
-      expect(prettifyNumber(0.0000001)).toEqual('< 0.000001');
-    });
+      test('should return "0.0₅1" for input 0.000001 and no currentCurrency selected', () => {
+        expect(prettifyNumber(0.000001)).toEqual('0.0₅1');
+      });
 
-    test('should return "1,321,965,595" for large number', () => {
-      expect(prettifyNumber(1321965595)).toEqual('1,321,965,595');
-    });
+      test('should return "0.0₆12345" for input 0.000000123456 and no currentCurrency selected', () => {
```

Comment:
I don't see tests for negative exponents (e-4), a case you are covering in the code

Reply:
All small number in the unit test are transformed into e-x by the SafeDecimal.toString().
I actually had to add this case because of unit test.
I think I can add test for the `subscript` function itself